### DISCUSSION TITLE
How to switch .NET MAUI Flyout of Shell to the right with FlowDirection?

### DISCUSSION THREAD

**Discussion by RafikMk** [dotnet/maui#15202] [Q&A]
Hello everyone,

I'm currently working on a .NET MAUI application and facing a challenge with the Flyout menu (Sidebar). I'm targeting a right-to-left layout to better cater to right-to-left languages.

I've attempted to use the FlowDirection="RightToLeft" attribute within the Shell, as shown below:
```
<Shell xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
       xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
       x:Class="MyApp.AppShell"
       FlowDirection="RightToLeft">
```
While this successfully reverses my content's flow, it doesn't affect the Flyout menu which still slides from the left. Instead, I'm aiming to have the Flyout emerge from the right.

Is there a way to achieve this in .NET MAUI?

Thanks in advance for your assistance.

**Comment by Xyncgas** [comment]
```
using Android.Content;
using Android.Support.Design.Widget;
using Android.Views;
using YourAppName;
using YourAppName.Droid.Renderers;
using Xamarin.Forms;
using Xamarin.Forms.Platform.Android;

[assembly: ExportRenderer(typeof(RTLFlowMasterDetailPage), typeof(RTLFlowMasterDetailPageRenderer))]
namespace YourAppName.Droid.Renderers
{
    public class RTLFlowMasterDetailPageRenderer : MasterDetailPageRenderer
    {
        public RTLFlowMasterDetailPageRenderer(Context context) : base(context)
        {
        }

        protected override void OnLayout(bool changed, int l, int t, int r, int b)
        {
            base.OnLayout(changed, l, t, r, b);

            var master = FindViewById<NavigationView>(Resource.Id.master);
            master.LayoutDirection = LayoutDirection.Rtl;
        }
    }
}
```

**Reply by RafikMk** [reply]
Bien sûr, vous pouvez répondre quelque chose comme ça :

Merci pour votre réponse rapide. Cependant, il semble que le code que vous avez partagé est spécifique à Android. Je cherche plutôt une solution pour une application Windows en .NET MAUI. Y a-t-il une manière d'accomplir cela pour Windows? Je vous remercie d'avance pour toute aide supplémentaire que vous pourriez offrir.
